community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/StoreUpgrader.java

```diff
@@ -116,8 +118,6 @@ public void migrateIfNeeded( File storeDirectory)
             throw new UpgradeNotAllowedByConfigurationException();
         }
 
-        CustomIOConfigValidator.assertCustomIOConfigNotUsed( config, CUSTOM_IO_EXCEPTION_MESSAGE );
-
```

Comment:
One argument against this is that the `StoreUpgrader` would need to become aware of store versions, instead of that being isolated to `StoreMigrator`